lib/usd/utils/MergePrims.cpp

```diff
@@ -159,6 +161,138 @@ template <class T> auto makeFuncWithContext(const MergeContext& ctx, T&& func)
         func, ctx, ph::_1, ph::_2, ph::_3, ph::_4, ph::_5, ph::_6, ph::_7, ph::_8, ph::_9);
 }
 
+//----------------------------------------------------------------------------------------------------------------------
+// Special transform attributes handling.
+//
+// The goal is to detect that a prim transform has *not* changed even though its representation
+// might have changed. For example it might use different attributes (RotateX vs RotateXYZ) or
+// different transform operations order. (These are in the xformOpOrder attribute.)
+//----------------------------------------------------------------------------------------------------------------------
+
+const std::set<TfToken>& getTransformAttributeNames()
+{
+    static const std::set<TfToken> trfAttrs = {
+        TfToken("translate"),
+        TfToken("scale"),
+        TfToken("orient"),
+        TfToken("rotateX"),
+        TfToken("rotateY"),
+        TfToken("rotateZ"),
+        TfToken("rotateXYZ"),
+        TfToken("rotateXZY"),
+        TfToken("rotateYXZ"),
+        TfToken("rotateYZX"),
+        TfToken("rotateZXY"),
+        TfToken("rotateZYX"),
+        TfToken("transform"),
+        TfToken("xformOp:translate"),
+        TfToken("xformOp:scale"),
+        TfToken("xformOp:orient"),
+        TfToken("xformOp:rotateX"),
+        TfToken("xformOp:rotateY"),
+        TfToken("xformOp:rotateZ"),
+        TfToken("xformOp:rotateXYZ"),
+        TfToken("xformOp:rotateXZY"),
+        TfToken("xformOp:rotateYXZ"),
+        TfToken("xformOp:rotateYZX"),
+        TfToken("xformOp:rotateZXY"),
+        TfToken("xformOp:rotateZYX"),
+        TfToken("xformOp:transform"),
+        TfToken("xformOpOrder"),
+    };
+
+    return trfAttrs;
+}
+
+bool isTransformProperty(const UsdProperty& prop)
+{
+    const auto& trfAttrNames = getTransformAttributeNames();
+    return trfAttrNames.count(prop.GetBaseName()) > 0;
+}
+
+std::vector<UsdAttribute> getTransformAttributes(const UsdPrim& prim)
+{
+    // To retrieve transform attributes, retrieve all attributes and then
+    // remove the non-transform ones. This is fast because we scan the
+    // attributes only once.
+    std::vector<UsdAttribute> trfAttrs = prim.GetAttributes();
+
+    const auto newEnd
+        = std::remove_if(trfAttrs.begin(), trfAttrs.end(), [](const UsdAttribute& attr) {
+              return !isTransformProperty(attr);
+          });
+    trfAttrs.erase(newEnd, trfAttrs.end());
+
+    return trfAttrs;
+}
+
+GfMatrix4d getLocalTransform(const UsdPrim prim, const UsdTimeCode& timeCode)
+{
+    UsdGeomXformable xformable(prim);
+
+    // Note: we don't check the return of GetLocalTransformation() because prims
+    //       can lack transform, like the root for example.
```

Comment:
If the prim doesn't support the UdsGeomXformable schema, you should return the identity:
if (!xformable) {
    return GfMatrix4d(1);
}

You should then be able to TF_VERIFY that GetLocalTransformation() returns true.

Reply:
Yes, the matrix is initialized with that for this reason, although I had assumed that if the function failed the variable passed in was not modified, which I guess I had no reason to believe it would always be true.